src/common/arrow/src/native/read/array/binary.rs

```diff
@@ -308,7 +309,18 @@ fn try_new_binary_array<O: Offset>(
     validity: Option<Bitmap>,
 ) -> Result<Box<dyn Array>> {
     if matches!(data_type, DataType::Utf8 | DataType::LargeUtf8) {
-        let array = Utf8Array::<O>::try_new(data_type, offsets, values, validity)?;
+        // todo!("new string")
```

Comment:
Looks like this `todo` can be deleted

Reply:
try_new will check utf8. i was planning to use try_new_unchecked afterwards.